tests/test_workers/test_tasks/test_general.py

```diff
@@ -259,25 +263,49 @@ def test_finalize_request(
     mock_create_archive,
     mock_get_request,
     task_passes_state_check,
+    retries_until_file_is_ready,
+    raise_error,
 ):
     pkg = {"name": "foo", "version": "1.0", "type": "pip"}
-    mock_get_request.return_value = {
-        "flags": ["some-flag"],
-        "pkg_managers": ["pip"],
-        "packages": [pkg],
-        "dependencies": [pkg, pkg],
-    }
 
+    # the mock will be called at least twice, once for the first loading of the request and
+    # once for the first check on the packages file
+    expected_mock_calls = retries_until_file_is_ready + 2
+
+    def side_effect(*args):
+        call_count = mock_get_request.call_count
+
+        # the mock should return a full answer only when we reach the expected amount of retries
+        if call_count == expected_mock_calls:
+            return {
+                "flags": ["some-flag"],
+                "pkg_managers": ["pip"],
+                "packages": [pkg],
+                "dependencies": [pkg, pkg],
+            }
+        else:
+            return {
+                "flags": ["some-flag"],
+                "pkg_managers": ["pip"],
+                "packages": [],
+                "dependencies": [],
+            }
+
+    mock_get_request.side_effect = side_effect
     mock_aggregate_data.return_value = mock.Mock(packages=[pkg], all_dependencies=[pkg, pkg])
 
-    tasks.finalize_request(42)
+    with raise_error and pytest.raises(CachitoError) or nullcontext():
```

Comment:
Stole this shamelessly from @serg-cymbaluk ;)

Reply:
Maybe check the expected error message here (with `raises(..., match="some message")`)?